fv/infrastructure/infra_etcd.go

```diff
@@ -97,7 +97,7 @@ func (eds *EtcdDatastoreInfra) AddNode(felix *Felix, idx int, needBGP bool) {
 	felixNode.Spec.IPv4VXLANTunnelAddr = felix.ExpectedVXLANTunnelAddr
 	if needBGP {
 		felixNode.Spec.BGP = &api.NodeBGPSpec{
-			IPv4Address:        felix.IP,
+			IPv4Address:        felix.IP + "/24",
```

Comment:
I'm guessing this is to trick the FVs into thinking its in the same subnet? 

Do we know what the actual subnet is? It would be much better if we could just use that rather than pretending it is a /24 when it might not actually be.

Reply:
I'm not sure, I'm still a little fuzzy on how some of this stuff works for the fv's

Reply:
Looks like the IP address is set from here:
- https://github.com/projectcalico/felix/blob/9891c8b588f38e19634f75e0dbb1174a6968a74e///fv/containers/containers.go#L193
- https://github.com/projectcalico/felix/blob/9891c8b588f38e19634f75e0dbb1174a6968a74e///fv/containers/containers.go#L297:21

We could add a new field, e.g. `IPPrefixLen` and add it to the `Container` Struct and then populate it from `docker inspect` output for the container (which has a field with the same name). 

Looking at Docker inspect output should show you what that the network looks like for your containers.

e.g., here's the network section for a container I'm running locally:

```
           "Networks": {
                "operator-test-cluster": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "k3d-operator-test-cluster-worker-0",
                        "31e922449d40"
                    ],
                    "NetworkID": "43f081b512f3e9ee012b2bba17384e983085665ff57629ced783861ebd00e801",
                    "EndpointID": "cdcb6bda08058d84cfe82483d0a74ab0d977812dcce29edafcb23c401116aff9",
                    "Gateway": "172.29.0.1",
                    "IPAddress": "172.29.0.3",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:1d:00:03",
                    "DriverOpts": null
                }
            }
```